src/v/redpanda/admin/transform.cc

```diff
@@ -250,13 +267,36 @@ admin_server::deploy_transform(std::unique_ptr<ss::http::request> req) {
         }
     }
 
+    model::transform_offset_options offset_opts{};
+    if (doc.HasMember("offset")) {
+        auto offset = doc["offset"].GetObject();
+        auto format = ss::sstring{
+          offset["format"].GetString(), offset["format"].GetStringLength()};
+        auto value = offset["value"].GetInt64();
+
+        if (format == "timestamp") {
+            using tp = model::timestamp_clock::time_point;
+            using dur = model::timestamp_clock::duration;
+            if (value >= dur::max() / 1ms) {
+                throw ss::httpd::bad_request_exception(fmt::format(
+                  "Bad offset: Timestamp value out of range ({})", value));
+            }
+            offset_opts.position = model::to_timestamp(tp{value * 1ms});
+        } else if (format == "from_start") {
+            offset_opts.position = model::offset{value};
+        } else if (format == "from_end") {
+            offset_opts.position = model::offset{-value};
+        }
+    }
```

Comment:
else 400?

Reply:
Strictly speaking, covered by the validator, but ya probably wise

Reply:
done